CodeStarterCamp_Week3/Step1.swift

```diff
@@ -9,49 +9,60 @@ import Foundation
 
 class Person {
     var name: String
-    var asset: Int = 0
+    var asset: Int
+    
     init(name: String, asset: Int) {
         self.name = name
         self.asset = asset
     }
     
-    func buyCoffee(coffeeShop: CoffeeShop, menus: [Coffee]) {
-        coffeeShop.getOrder(menus: menus)
-        for menu in menus {
-            if let price = coffeeShop.menus[menu.rawValue] {
+    func buyCoffee(menu: [Coffee], at coffeeShop: CoffeeShop) {
+        coffeeShop.takeOrder(menu: menu)
+        for coffee in menu {
+            if let price = coffeeShop.menu[coffee] {
                 self.asset -= price
             }
         }
     }
-    
 }
 
 class CoffeeShop {
     var barista: Person?
     var sales: Int = 0
-    var menus: Dictionary<String, Int> = [:]
-    var pickUpTable: Array<String> = []
-    init() {
-        self.menus = [Coffee.americano.rawValue: 4000,
-                      Coffee.latte.rawValue: 4500,
-                      Coffee.iceTea.rawValue: 5000]
+    var menu: [Coffee: Int]
+    var pickUpTable: [Coffee] = []
+    
+    init(menu: [Coffee: Int]) {
+        self.menu = menu
     }
-    func getOrder(menus: [Coffee]) {
-        for menu in menus {
-            self.makeCoffee(menu.rawValue)
+    
+    func takeOrder(menu: [Coffee]) {
+        for coffee in menu {
+            self.makeCoffee(coffee)
         }
     }
     
-    func makeCoffee(_ menu: String) {
-        self.pickUpTable.append(menu)
-        if let sale = menus[menu] {
+    func makeCoffee(_ coffee: Coffee) {
+        self.pickUpTable.append(coffee)
+        if let sale = menu[coffee] {
             self.sales += sale
         }
     }
 }
 
 enum Coffee: String {
-    case americano = "americano"
-    case latte = "latte"
-    case iceTea = "iceTea"
+    case americano
+    case latte
+    case iceTea
+    
+    func price() -> Int {
+        switch self {
+        case .americano:
+            return 4000
+        case .latte:
+            return 4500
+        case .iceTea:
+            return 4500
+        }
+    }
```

Comment:
price 함수를 만들어주셨군요..! 좋습니다 👍
연산프로퍼티로 더 간단하게 나타낼 수 있답니다~ 👀
```suggestion
    var price: Int {
        switch self {
        case .americano:
            return 4000
        case .latte:
            return 4500
        case .iceTea:
            return 4500
        }
    }
```

이러면 호출할 때 괄호를 넣지 않아도 되겠죠?!
```swift
let americanoPrice = CoffeeShop.americano.price
```